h2o-py/tests/testdir_algos/glrm/pyunit_glrm_PUBDEV_3728_binary_numeric.py

```diff
@@ -0,0 +1,104 @@
+                                          seed=12345, init="SVD")
+  try:
+    glrm_h2o.train(x=prostateF.names, training_frame=prostateF, validation_frame=prostateF)
+    assert False, "Should encounter error when with binary loss with categorical loss function and init=SVD for GLRM."
+  except Exception as e:
+    print(e)
+    if sys.version_info < (3,0,0):  # use for python version 2.
+        assert "Cannot use init mode SVD when using binary" in e.message.exception_msg, \
+            "Should have raised error concerning using init mode SVD with GLRM, binary loss with binary categorical " \
+            "columns."
+    else:
+        assert "Cannot use init mode SVD when using binary" in e.args[0]._props["exception_msg"], \
+            "Should have raised error concerning using init mode SVD with GLRM, binary loss with binary categorical " \
+            "columns."
+
+  # exercise logistic loss with numeric columns
+  glrm_h2o_num = H2OGeneralizedLowRankEstimator(k=5, loss_by_col=loss_all, recover_svd=True, transform="STANDARDIZE",
+                                              seed=12345, init="SVD")
+  glrm_h2o_num.train(x=prostateF_num.names, training_frame=prostateF_num, validation_frame=prostateF_num)
+  glrm_h2o_num.show()
+
+  print("check with init = user")
+  initial_y = [[-1.27675647831893E-15,64.87421383647799,2.0,1.0,2.0816681711721685E-16,8.533270440251574,
+                9.380440251572328,5.886792452830188],
+               [0.7297297297297298,66.05405405405405,2.0,0.0,1.0,23.270270270270274,9.589189189189193,7.27027027027027],
+               [0.01754385964912314,70.35087719298245,2.0,1.0,-1.3877787807814457E-17,10.078947368421053,
+                42.37543859649123,6.157894736842105],
+               [0.9,65.95,2.0,0.0,0.2,81.94500000000001,16.375,7.4],
+               [0.9999999999999989,65.48598130841121,2.0,3.0,1.3877787807814457E-16,13.3092523364486,
+                13.268411214953275,6.747663551401869]]
+  initial_y_h2o = h2o.H2OFrame(list(initial_y))
+  glrm_h2o = H2OGeneralizedLowRankEstimator(k=5, loss_by_col=loss_all, recover_svd=True, transform="STANDARDIZE",
+                                            seed=12345, init="User", user_y=initial_y_h2o)
+  glrm_h2o.train(x=prostateF.names, training_frame=prostateF, validation_frame=prostateF)
+  glrm_h2o.show()
+
+  # exercise logistic loss with numeric columns
+  glrm_h2o_num = H2OGeneralizedLowRankEstimator(k=5, loss_by_col=loss_all, recover_svd=True, transform="STANDARDIZE",
+                                              seed=12345, init="User", user_y=initial_y_h2o)
+  glrm_h2o_num.train(x=prostateF_num.names, training_frame=prostateF_num, validation_frame=prostateF_num)
+  glrm_h2o_num.show()
+
+  # singular values from glrm models should equal if binary columns with binary loss are read in as either
+  # categorical or numerics.  If not, something is wrong.
+  assert pyunit_utils.equal_two_arrays(glrm_h2o._model_json["output"]["singular_vals"],
```

Comment:
It looks like this is comparing the same model to itself (both are `glrm_h2o`).  I think one should be changed to `glrm_h2o_num`.